lib/matplotlib/tests/test_colorbar.py

```diff
@@ -827,3 +827,30 @@ def test_aspects():
     np.testing.assert_almost_equal(
         cb[1][0].ax.get_position(original=False).height * 2,
         cb[1][2].ax.get_position(original=False).height, decimal=2)
+
+
+@image_comparison(['proportional_colorbars.png'], remove_text=True,
+                  style='mpl20')
+def test_proportional_colorbars():
+
+    x = y = np.arange(-3.0, 3.01, 0.025)
+    X, Y = np.meshgrid(x, y)
+    Z1 = np.exp(-X**2 - Y**2)
+    Z2 = np.exp(-(X - 1)**2 - (Y - 1)**2)
+    Z = (Z1 - Z2) * 2
+
+    levels = [-1.25, -0.5, -0.125, 0.125, 0.5, 1.25]
+
+    cmap = mcolors.ListedColormap(['r', 'g', 'white', 'g', 'r'])
+    cmap.set_under('yellow')
+    cmap.set_over('cyan')
```

Comment:
Optional, but the test is stonger if you break the symmetry of the colors. With the symmetry, you cannot check the sign.

```suggestion
    cmap = mcolors.ListedColormap(
        ['0.3', '0.5', 'white', 'lightblue', 'steelblue'])
    cmap.set_under('darkred')
    cmap.set_over('crimson')
```

![grafik](https://user-images.githubusercontent.com/2836374/132060436-2e56005f-3f06-44ef-b5b7-55a5cf9c8e47.png)

Also, I've chosen colors that hurt the eye a little less :smile:

Reply:
sure